lib/src/layer/tile_layer/tile_coordinates.dart

```diff
@@ -11,13 +11,10 @@ class TileCoordinates extends Point<int> {
   @override
   String toString() => 'TileCoordinate($x, $y, $z)';
 
-  // Overridden because Point's distanceTo does not allow comparing with a point
-  // of a different type.
-  @override
-  double distanceTo(Point<num> other) {
+  double distanceToSq(Point<double> other) {
     final dx = x - other.x;
     final dy = y - other.y;
-    return sqrt(dx * dx + dy * dy);
+    return dx * dx + dy * dy;
   }
```

Comment:
Not 100% a fan of this. Would prefer it if there was a separate `distanceToSq` method for more efficient internal use.

Reply:
No strong feelings. More importantly are you happy with me removing the distanceTo overload? :)

Reply:
Not really, feels like an unnecessary breaking change.

Reply:
I understand. I'm trying to be opportunistic here. It rubs me that TileCoordinate is a Point<int> (which semantically it is not) and then overrides Point<int>'s with even more error prone behaviors. Is there a process for us to clean these up or at least stop the bleeding?

Reply:
Ok, let's keep this PR non-breaking. Then in a separate PR, we can do breaking changes, with as many deprecations as possible.

Reply:
@ignatz You know what, scratch that :D. We're trying to improve performance here, so breaking changes are needed. I've approved and merged your other PR with breaking changes as well.
However, in this case, I still think `distanceTo` can be useful, and so shouldn't be removed.

Apologies for the messing about, but if you're serious about not liking `Point` and it's generic types, I'll allow a PR with breaking changes.

Reply:
No worries at all. I understand the motivations. I'm also been hit by keeping in sync with some of the v3 -> v6 major changes. I'm mostly just keeping the nag-rate high :)

Reply:
>  However, in this case, I still think distanceTo can be useful, and so shouldn't be removed.

(I put it back. We can always defer breaking changes until v7. I do disagree though that one should support distance calculations between different types of points. Chances are it's accidental. I do think it's fair and more explicit to ask users to convert a coordinate to a Point<double> first. Either way, I'm not going to loose any sleep over this. I just think that FM will have an easier time long-term if from a separation of responsibilities point of view, it's not in the business of providing convenience on top (or extending) external types.... I'll shut up now :) )